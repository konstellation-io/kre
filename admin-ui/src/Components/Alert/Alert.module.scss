@import '../../styles/colors';
@import '../../styles/mixins';
@import '../../styles/shadows';

.container {
  @include shadow(4);

  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
  height: 60px;
  margin-bottom: 3px;
  padding: 0 20px;
  background-color: palette(error, 900);
  white-space: nowrap;
}

.label {
  @include font-button;
  color: $color-black;

  width: 105px;
  height: 24px;
  padding: 0 5px;
  line-height: 24px;
  text-align: center;

  &.error {
    background-color: palette(error);
  }
}

.message {
  @include font-body;
  font-weight: 600;
  color: white;

  padding: 0 20px;
  margin-right: auto;
  overflow: hidden;
  text-overflow: ellipsis;
}
