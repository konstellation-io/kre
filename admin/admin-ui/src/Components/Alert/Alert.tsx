import Button from '../Button/Button';
import ROUTE from 'Constants/routes';
import React from 'react';
import { buildRoute } from 'Utils/routes';
import cx from 'classnames';
import styles from './Alert.module.scss';

export type Props = {
  type: string;
  message: string;
  runtimeId: string;
};

function Alert({ type, message, runtimeId }: Props) {
  return (
    <div className={styles.container}>
      <div className={cx(styles.label, styles[type.toLowerCase()])}>{type}</div>
      <div className={styles.message} title={message}>
        {message}
      </div>
      <Button
        label="GO TO RUNTIME"
        to={buildRoute.runtime(ROUTE.RUNTIME, runtimeId)}
      />
    </div>
  );
}

export default Alert;
