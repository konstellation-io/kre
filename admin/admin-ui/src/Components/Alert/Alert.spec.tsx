import Alert from './Alert';
import { Button } from 'kwc';
import React from 'react';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Alert type="ERROR" message="some message" runtimeId="runtimeID" />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('shows right components', () => {
    expect(wrapper.exists(Button)).toBeTruthy();
    expect(wrapper.find('.message').text()).toBe('some message');
  });
});
