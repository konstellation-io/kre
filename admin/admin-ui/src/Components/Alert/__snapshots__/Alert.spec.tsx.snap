// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="label error"
  >
    ERROR
  </div>
  <div
    className="message"
    title="some message"
  >
    some message
  </div>
  <A
    label="GO TO RUNTIME"
    to="/runtimes/runtimeID"
  />
</div>
`;
